packages/venia-ui/lib/components/Header/cartTrigger.js

```diff
@@ -1,75 +1,67 @@
-import React, { Component } from 'react';
-import { connect } from '@magento/venia-drivers';
-import { compose } from 'redux';
-import PropTypes from 'prop-types';
+import React, { useEffect, useMemo } from 'react';
+import { func, number, shape, string } from 'prop-types';
+import { ShoppingCart as ShoppingCartIcon } from 'react-feather';
 
-import { toggleCart } from '../../actions/cart';
+import Icon from '../Icon';
 import CartCounter from './cartCounter';
 
-import Icon from '../Icon';
-import { ShoppingCart as ShoppingCartIcon } from 'react-feather';
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import defaultClasses from './cartTrigger.css';
 
-export class Trigger extends Component {
-    static propTypes = {
-        children: PropTypes.node,
-        classes: PropTypes.shape({
-            root: PropTypes.string
-        }),
-        toggleCart: PropTypes.func.isRequired,
-        itemsQty: PropTypes.number
+const getCartIcon = cartDetails => {
+    const itemsQty = cartDetails.items_qty;
+    const iconColor = 'rgb(var(--venia-text))';
+    const svgAttributes = {
+        stroke: iconColor
     };
 
-    get cartIcon() {
-        const {
-            cart: { details }
-        } = this.props;
-        const itemsQty = details.items_qty;
-        const iconColor = 'rgb(var(--venia-text))';
-        const svgAttributes = {
-            stroke: iconColor
-        };
+    if (itemsQty > 0) {
+        svgAttributes.fill = iconColor;
+    }
 
-        if (itemsQty > 0) {
-            svgAttributes.fill = iconColor;
-        }
+    return <Icon src={ShoppingCartIcon} attrs={svgAttributes} />;
+};
 
-        return <Icon src={ShoppingCartIcon} attrs={svgAttributes} />;
-    }
+const Trigger = props => {
+    const { cart, getCartDetails, toggleCart } = props;
+    const { details: cartDetails } = cart;
+    const { items_qty: numItems } = cartDetails;
 
-    render() {
-        const {
-            classes,
-            toggleCart,
-            cart: { details }
-        } = this.props;
-        const { cartIcon } = this;
-        const itemsQty = details.items_qty;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-        return (
-            <button
-                className={classes.root}
-                aria-label="Toggle mini cart"
-                onClick={toggleCart}
-            >
-                {cartIcon}
-                <CartCounter counter={itemsQty ? itemsQty : 0} />
-            </button>
-        );
-    }
-}
+    useEffect(() => {
+        if (getCartDetails) {
+            getCartDetails();
+        }
+    }, [getCartDetails]);
+
+    const cartIcon = useMemo(() => {
+        return getCartIcon(cartDetails);
```

Comment:
Minor improvement would be to pull off the `items_qty` from `cartDetails` as it seems like the only value the function cares about.

Beyond that we could also just hardcode the two possible icon states, one that has color and one that is filled and one that isn't.

Reply:
Resolved by https://github.com/magento/pwa-studio/pull/1556/commits/ced668878ed9b61219559142c28ecbfc2bff75d4.